core/src/unprocessed_packet_batches.rs

```diff
@@ -368,9 +374,40 @@ pub fn packet_message(packet: &Packet) -> Result<&[u8], DeserializedPacketError>
         .ok_or(DeserializedPacketError::SignatureOverflowed(sig_size))
 }
 
-/// Computes `(addition_fee + base_fee / requested_cu)` for `deserialized_packet`
-fn compute_fee_per_cu(_message: &VersionedMessage, _bank: &Bank) -> u64 {
-    1
+/// Computes `(addition_fee / requested_cu)` if feature tx_wide_compute_cap is enabled,
+/// otherwise return default 0u64
+fn compute_fee_per_cu(message: &VersionedMessage, bank: &Bank) -> u64 {
+    if bank.feature_set.is_active(&tx_wide_compute_cap::id()) {
+        if let Some(sanitized_message) = sanitize_message(message, bank) {
+            let mut compute_budget = ComputeBudget::default();
+            match compute_budget.process_message(&sanitized_message, false, true) {
+                Ok(requested_additional_fee) => requested_additional_fee / compute_budget.max_units,
```

Comment:
Running tests with `request_additional_fee = 3`, but the prioritization didn't happen, only to realize fee-per-cu would be `0` if requested additional_fee is less than requested units, which is 200K by default. So fee-per-cu needs to be float (don't like float), or scale up (risk overflow). debating

Reply:
what are the units currently for `requested_additional_fee`? If it's using lamports it should be much larger than the default compute units. 

We can also just multiply all the fees by a constant magnifying factor (the max compute units for instance) to account for any future fluctuations in SOL price

Reply:
If we do scale up, we should also add a test here if we're using constants like the max compute units that we can never overflow.

Reply:
Since it is fee-per-cu, I am assuming it is in lamports, as in "how many lamports per cu". Can use inputs from @jackcmay. `Additional_fee` is defined as u32 has MAX `4294967295` for a transaction, if the transaction use default 200,000 units, when user can bid up to 21,474 lamports per unit, that may not be enough?

Reply:
Additional fee is the total additional lamports the payer is willing to pay, it is separate from CU requested. 

There Is a mechanism in the bank to calculate a base fee based on the number of compute units requested and right now it's set to match the current per-sig fee.  That can be changed anytime with a feature gate to any quantized fee per n units.  The total fee for the tx would then be the fee per n units + additional fee

Reply:
we can prioritize based on additional_fee only.  I think it might be worth it to set the base fee to some fixed amount thats the same for all txs.

Reply:
@jackcmay since we are prioritize only by additional_fee, what do you think to re-define `additional_fee` to `additional_lamports_per_cu`? Would make it cleaner?

Reply:
What if we change additional_fee to u64 and call it additional_fee_lamports?  Would require a change to the compute budget instruction and would delay the enabling of tx-wide cap on mnb...

I'm also fine with additional_lamports_per_cu, if so, should rename it in the instruction

Reply:
Yea, both will works. Am debating with @aeyakovenko about it. 

With first one, we will have to deal with precision issues come with floats (eg, 2 TXs both has 1.4M CUs, one has additional_fee of `1`, other has `2`, comparing their fee-per-cu wouldn't be accurate, but might be OK). 

With second option, if we want the smallest fee bump to be less than 1 lamports-per-cu, say 0.001 lamport-per-cu, we then need some multiplier / factor. Could add confusion to user. 

Let's take a vote 😃

Reply:
a difference of 1 or 2 lamports is pretty small difference

Reply:
How about this?
https://github.com/solana-labs/solana/pull/25027

Reply:
> a difference of 1 or 2 lamports is pretty small difference

just an example 😜

Reply:
> How about this? #25027

Looks good, can take `u64` approach

Reply:
f64's for fee-per-cu seem fine?

f64 has 15 decimal precision, so on a 1.4m compute unit transaction you would only need 1.4 * 10^-9 lamports increase on your total fee to move the fee-per-cu. This is well below the 1 lamport incremental units specified in the additional fee

Reply:
commit b2c17755d887e0c5bcd0b375c25260997c6e3bb9 to change fee-per-cu to f64.

Reply:
am i the only one concerned that prioritizing based purely on `additional_fee` is going to create perverse incentives?  they encode nothing except, "user desire to execute a transaction," which has nothing to do with resource contention within the system.  that is, users have to pay the price for _every_ resource under contention, regardless of whether their transaction makes use of it.  otherwise independent resource contentions become additive, everyone loses.

this seems to me the opposite of what the cluster economics should be trying to do, which is to minimize negative externalities of resource contention. containing fee rise to individual resource puts pressure on dapps to optimize their code and/or institute internal economic games. a Good Thing :tm:.  i recognize that we aren't in a position to implement anything like this today, but i don't want to see its possibility painted out of the picture out of desperation.

never mind there being
* zero client-side support for `additional_fee`, basically handing the chain over to (bot) devs in the meantime
* no tooling to inform effective `additional_fee` values
* a fairly non-trivial to resolve [footgun](https://github.com/solana-labs/solana/issues/24612) stemming from the fact that `addition_fee` is under the transaction hash, preventing re-submission of the otherwise same transaction with a higher fee without jumping through hoops or risking both submissions being executed.

Reply:
@aeyakovenko @carllin what's your take on @t-nelson's concern?

Using `additional_fee / requested_cu` to prioritize transactions gives user a tool to increase chance of landing their TXs regardless if there's resource contention, but especially useful during contention. Believe it's a real possibility that bots takes advantage of this before real user, but the fee should discourage bots going too far, would it?

Reply:
> Additional fee is the total additional lamports the payer is willing to pay, it is separate from CU requested.
> 
> There Is a mechanism in the bank to calculate a base fee based on the number of compute units requested and right now it's set to match the current per-sig fee. That can be changed anytime with a feature gate to any quantized fee per n units. The total fee for the tx would then be the fee per n units + additional fee

Hmm might it be easier if it's just price of n units * whole number fee multiplier? I feel like having an additional fee that can be incremented to a tiny degree could result in a game between bots trying to minutely increase fee priority that results in an increase in spam. It might also be easier for UX because users could choose fee multiplier defaults in the wallet.

Reply:
the `whole number multiplier` sounds similar to `additional_lamports_per_cu` suggested [early](https://github.com/solana-labs/solana/pull/25007#discussion_r866363339). I like this idea for its simplicity.